ndcube/ndcube.py

```diff
@@ -463,24 +463,19 @@ def crop_by_coords(self, lower_corner, interval_widths=None, upper_corner=None,
                 raise TypeError("lower_corner and interval_widths/upper_corner must be "
                                 "of type astropy.units.Quantity or the units kwarg "
                                 "must be set.")
-        # Derive all corners coordinates
-        quantity_list = [[lower_corner[i], upper_corner[i]] for i in range(n_dim)]
-        all_corners = [self.world_to_pixel(*a) for a in product(*quantity_list)]
-        # Inputing of all corners coordinates inside a numpy array
-        # According to the boundary conditions
-        corners_array = np.zeros((2**n_dim, n_dim))
-        for i in range(2**n_dim):
-            for j in range(n_dim):
-                corners_array[i, j] = u.Quantity(all_corners[i][j]).value
-                if corners_array[i, j] > self.data.shape[j]:
-                    corners_array[i, j] = self.data.shape[j]
-                if corners_array[i, j] < 0:
-                    corners_array[i, j] = 0
-        # Taking the maximum and minimum values of coordinates
-        lower_pixels = corners_array.min(0)
-        upper_pixels = corners_array.max(0)
-        # Creating a tuple to crop the data with inputed coordinates
-        item = tuple([slice(int(lower_pixels[i]), int(upper_pixels[i]) + 1) for i in range(n_dim)])
+        # Get all corners of region of interest.
+        all_world_corners_grid = np.meshgrid(*[u.Quantity([lower_corner[i], upper_corner[i]],
+                                                          unit=lower_corner[i].unit).value
+                                               for i in range(self.data.ndim)])
```

Comment:
Suggestion for code readability:
```python
all_world_corners_grid = np.meshgrid(
    *[u.Quantity([lower_corner[i], upper_corner[i]], unit=lower_corner[i].unit).value
      for i in range(self.data.ndim)]
```
It makes the code more readable but minimizing the break up of single items (e.g. Quantities) over multiple lines.

Reply:
ping @BaptistePellorceAstro.  This comment got outdated incorrectly.

Reply:
Ok I was thinking about it ! I am solving some test bugs from the last commits.